libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/MapboxExtendableButtonParams.kt

```diff
@@ -0,0 +1,50 @@
+package com.mapbox.navigation.dropin
+
+import android.widget.LinearLayout
+import androidx.annotation.StyleRes
+
+/**
+ * Params used to modify button appearance.
+ *
+ * @param style style res of a button.
+ * @param layoutParams layout params of a button.
+ */
+class MapboxExtendableButtonParams(
+    @StyleRes
+    val style: Int,
+    val layoutParams: LinearLayout.LayoutParams,
```

Comment:
With `LinearLayout.LayoutParams` we can not apply margins to `layoutParams`, but also apply other properties like `gravity`. Can we do the same with `ViewGroup.MarginLayoutParams`?
@Zayankovsky

Reply:
but we need somehow pass `width/height`
we can pass them as `Int`, wdyt?

Reply:
You can't apply gravity with `MarginLayoutParams`. If we want to support gravity customization, then we can use `LinearLayout.LayoutParams` like suggested or create our own class that contains only properties we would like to support.

Reply:
`MarginLayoutParams` supports width and height as well as margins.

Reply:
`gravity` might be useful if buttons have different size, so we need to move them to the start/end of the parent

@abhishek1508 wdyt?

Reply:
```
class MapboxExtendableButtonParams(
    val style: Int,
    val layoutParams: MarginLayoutParams,
    val gravity: Int
}
```

Reply:
Problem with gravity is evident in the example at present. If you look at the video, you will see that the circular action buttons are not centered horizontally.
<img width="200" alt="Screen Shot 2022-09-15 at 2 32 52 PM" src="https://user-images.githubusercontent.com/9770186/190362954-5d21b607-3b0b-4865-b680-ea5dcda36d36.png">

>so we need to move them to the start/end of the parent

I did not get your question. Can you please elaborate? @korshaknn

Reply:
@abhishek1508 I mean that we need to pass `gravity` to fix such issues ^^, so I see 2 options:
- pass `LinearLayout.LayoutParams` (it has `gravity` option)
- pass `MarginLayoutParams ` and extra `gravity: Int ` option

Reply:
> ```
> class MapboxExtendableButtonParams(
>     val style: Int,
>     val layoutParams: MarginLayoutParams,
>     val gravity: Int
> }
> ```

I am not a huge fan of it. We will be basically adding wrapper to support properties that `LayoutParams` support by default. These sort of features are usually requested by users historically and I feel we will get requests for other properties support. 

> What if we decide to use another container in the future, e.g. ConstraintLayout? Should we use ViewGroup.MarginLayoutParams instead?

This is a valid concern, but doing this means we will be breaking SEMVER. These sort of changes can happen only in next major release. I would ideally not worry about changing from using `LinearLayout` to some other layout in our `.xml` file.

cc @korshaknn @Zayankovsky

Reply:
different gravity for each action button ^^

Reply:
I am suggesting that we should leave it as `LinearLayout.LayoutParams` because it is not gonna change unless we release next major version. We can still assign different gravity to the button using `LinearLayout.LayoutParams`; correct? @korshaknn

Reply:
I don't think changing the layout means breaking SEMVER, because this is not exposed to customers in any way.

Reply:
>  We can still assign different gravity to the button using LinearLayout.LayoutParams; correct?

yes, I've added a screenshot with such setup

Reply:
Anyway, I don't have strong opinions against `LinearLayout.LayoutParams`, if you think we are not going to change the container anytime soon.

Reply:
>Anyway, I don't have strong opinions against LinearLayout.LayoutParams, if you think we are not going to change the container anytime soon.

Resolving in favor of `LinearLayout.LayoutParams` @korshaknn